include/spdlog/sinks/qt_sinks.h

```diff
@@ -4,122 +4,81 @@
+  QObject *qt_object_ = nullptr;
+  std::string meta_method_;
 };
 
 #include "spdlog/details/null_mutex.h"
 #include <mutex>
-using qtextedit_sink_mt = qtextedit_sink<std::mutex>;
-using qtextedit_sink_st = qtextedit_sink<spdlog::details::null_mutex>;
-
-using qplaintextedit_sink_mt = qplaintextedit_sink<std::mutex>;
-using qplaintextedit_sink_st = qplaintextedit_sink<spdlog::details::null_mutex>;
-
+using qt_sink_mt = qt_sink<std::mutex>;
+using qt_sink_st = qt_sink<spdlog::details::null_mutex>;
 } // namespace sinks
 
 //
 // Factory functions
 //
-template <typename Factory = spdlog::synchronous_factory>
-inline std::shared_ptr<logger>
-qtextedit_logger_mt(const std::string &logger_name,
-                    QTextEdit *qtextedit = nullptr) {
-  return Factory::template create<sinks::qtextedit_sink_mt>(logger_name,
-                                                            qtextedit);
-}
-
-template <typename Factory = spdlog::synchronous_factory>
-inline std::shared_ptr<logger>
-qtextedit_logger_st(const std::string &logger_name,
-                    QTextEdit *qtextedit = nullptr) {
-  return Factory::template create<sinks::qtextedit_sink_st>(logger_name,
-                                                            qtextedit);
-}
-
-template <typename Factory = spdlog::synchronous_factory>
+template <typename T, typename Factory = spdlog::synchronous_factory>
 inline std::shared_ptr<logger>
-qplaintextedit_logger_mt(const std::string &logger_name,
-                    QPlainTextEdit *qplaintextedit = nullptr) {
-    return Factory::template create<sinks::qplaintextedit_sink_mt>(logger_name,
-                                                              qplaintextedit);
+qt_logger_mt(const std::string &logger_name, T *qt_object = nullptr, const std::string &meta_method = "") {
+  std::string meta_method_ = meta_method;
+  if (std::is_base_of<QTextEdit, T>::value)
+    meta_method_ = "append";
+  else if (std::is_same<T, QPlainTextEdit>::value)
+    meta_method_ = "appendPlainText";
+  return Factory::template create<sinks::qt_sink_mt>(logger_name, qt_object, meta_method_);
 }
 
-template <typename Factory = spdlog::synchronous_factory>
+template <typename T, typename Factory = spdlog::synchronous_factory>
 inline std::shared_ptr<logger>
-qplaintextedit_logger_st(const std::string &logger_name,
-                    QPlainTextEdit *qplaintextedit = nullptr) {
-    return Factory::template create<sinks::qplaintextedit_sink_st>(logger_name,
-                                                              qplaintextedit);
+qt_logger_st(const std::string &logger_name, T *qt_object = nullptr, const std::string &meta_method = "") {
+  std::string meta_method_ = meta_method;
+  if (std::is_base_of<QTextEdit, T>::value)
+    meta_method_ = "append";
+  else if (std::is_same<T, QPlainTextEdit>::value)
+    meta_method_ = "appendPlainText";
+  return Factory::template create<sinks::qt_sink_st>(logger_name, qt_object, meta_method_);
```

Comment:
T is known at compile time, so template specialization should be used instead of the if statements.

Reply:
I will convert solution with template specialization